### PR TITLE
Added search(-ms)/WebDAV rules

### PR DESCRIPTION
**Pull request by mbabinski**
### Summary of the Pull Request
Added rules based on recent research from Trellix and my own lab on ongoing exploitation of search/search-ms URI protocols to trick victims into running malicious lnk files hosted on WebDAV servers.

### Detailed Description of the Pull Request / Additional Comments
This PR adds three new rules:
1. WebDAV Temporary Local File Creation
2. Suspicious WebDAV LNK Execution
3. Search-ms and WebDAV Indicators in URL

These rules build on the work shared by Trellix here: https://www.trellix.com/en-us/about/newsroom/stories/research/beyond-file-search-a-novel-method.html. I've also used this narrower WebDAV-related rule as inspiration: https://github.com/SigmaHQ/sigma/blob/master/rules/windows/process_creation/proc_creation_win_cmd_net_use_and_exec_combo.yml.

### Example Log Event
For example logs and further details, see my recent blog post:
https://micahbabinski.medium.com/search-ms-webdav-and-chill-99c5b23ac462.

### CODE REVIEW

**Review by nasbench — APPROVED** [review]

File: rules/web/proxy_generic/proxy_webdav_search_ms.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            - 'agreement'
+            - 'invoice'
+            - 'notice'
+            - 'payment'
+    filter_main_local_ips:
+        dst_ip|cidr:
+            - '127.0.0.0/8'
+            - '10.0.0.0/8'
+            - '172.16.0.0/12'
+            - '192.168.0.0/16'
```
Comment (phantinuss) [inline rules/web/proxy_generic/proxy_webdav_search_ms.yml:34]:
```suggestion
    filter_main_local_ips:
        dst_ip|cidr:
            - '127.0.0.0/8'
            - '10.0.0.0/8'
            - '172.16.0.0/12'
            - '192.168.0.0/16'
            - '::1'  # IPv6 loopback
            - 'fe80::/8'  # IPv6 link-local addesses
            - 'fc00::/7'  # IPv6 private addresses
```
First time use of the cidr modifier for IPv6 addresses. Let's see what brakes :)

File: rules/windows/process_creation/proc_creation_win_webdav_lnk_execution.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+title: Potentially Suspicious WebDAV LNK Execution
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_webdav_lnk_execution.yml:1]:
```suggestion
title: Suspicious WebDAV LNK Execution
```
for a high rule we don't put a "potentially", do we?

Comment (nasbench) [reply rules/windows/process_creation/proc_creation_win_webdav_lnk_execution.yml:1]: I put it there and should be medium

File: rules/windows/process_creation/proc_creation_win_webdav_lnk_execution.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            - '\cscript.exe'
+            - '\mshta.exe'
+            - '\powershell.exe'
+            - '\pwsh.exe'
+            - '\wscript.exe'
+        CommandLine|contains: '\DavWWWRoot\'
+    condition: selection
+falsepositives:
+    - Unknown
+level: high
```
Comment (nasbench) [inline rules/windows/process_creation/proc_creation_win_webdav_lnk_execution.yml:34]:
```suggestion
level: medium
```

File: rules/web/proxy_generic/proxy_webdav_search_ms.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            - '::1'  # IPv6 loopback
+            - 'fe80::/8'  # IPv6 link-local addesses
```
Comment (phantinuss) [inline rules/web/proxy_generic/proxy_webdav_search_ms.yml:36]:
```suggestion
            - 'fe80::/8'  # IPv6 link-local addresses
```

File: rules/windows/process_creation/proc_creation_win_webdav_lnk_execution.yml (hunk 2 above)
Comment (phantinuss) [reply rules/windows/process_creation/proc_creation_win_webdav_lnk_execution.yml:1]: yeah or that way round

**Review by phantinuss — APPROVED** [review]

File: rules/web/proxy_generic/proxy_webdav_search_ms.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+            - 'notice'
+            - 'payment'
+    filter_main_local_ips:
+        dst_ip|cidr:
+            - '127.0.0.0/8'
+            - '10.0.0.0/8'
+            - '172.16.0.0/12'
+            - '192.168.0.0/16'
+            - '::1'  # IPv6 loopback
+            - 'fe80::/8'  # IPv6 link-local addresses
```
Comment (phantinuss) [inline rules/web/proxy_generic/proxy_webdav_search_ms.yml:36]:
```suggestion
            - 'fe80::/10'  # IPv6 link-local addresses
```

**Pull request merged**